### PR TITLE
Refactor Benchmarks and update calls that use `T::Swap` (Honzon and Cdp Treasury)

### PR DESCRIPTION
**Pull request by ferrell-code**

### CODE REVIEW

File: runtime/mandala/src/benchmarking/prices.rs [hunk 1]
```diff
@@ -16,28 +16,26 @@
...
-const STAKING: CurrencyId = GetStableCurrencyId::get();
-
 runtime_benchmarks! {
 	{ Runtime, module_prices }
 
 	lock_price {
 		// feed price
-		feed_price(vec![(STAKING, dollar(STAKING).into())])?;
-	}: _(RawOrigin::Root, STAKING)
+		feed_price(vec![(STABLECOIN, dollar(STABLECOIN).into())])?;
```
Comment (xlc) [inline runtime/mandala/src/benchmarking/prices.rs:31]: why change to Stablecoins? we have some special handle logic for Stablecoins in prices module so better to still use staking here

File: runtime/mandala/src/benchmarking/utils.rs [hunk 2]
```diff
@@ -98,6 +107,148 @@ pub fn dollar(currency_id: CurrencyId) -> Balance {
...
+	)?;
+	Ok(())
+}
+
+pub fn register_stable_asset() -> DispatchResult {
+	let asset_metadata = AssetMetadata {
+		name: b"Token Name".to_vec(),
+		symbol: b"TN".to_vec(),
+		decimals: 12,
+		minimal_balance: 1,
```
Comment (xlc) [inline runtime/mandala/src/benchmarking/utils.rs:149]:
```suggestion
		minimal_balance: 1_000_000_000,
```
so it is bit more realistic

File: runtime/mandala/src/benchmarking/prices.rs (hunk 1 above)
Comment (ferrell-code) [reply runtime/mandala/src/benchmarking/prices.rs:31]: I actually kept it the same look at line 26

```rust

const STAKING: CurrencyId = GetStableCurrencyId::get();
```

I can change it to staking cause yeah using stablecoin seems a bit nonsensical

Comment (xlc) [reply runtime/mandala/src/benchmarking/prices.rs:31]: oh ok that must be a mistake 😓

**Review by xlc — APPROVED** [review]

**Pull request merged**